identity/handler.go

```diff
@@ -44,6 +48,90 @@ func (h *Handler) RegisterAdminRoutes(admin *x.RouterAdmin) {
 
 	admin.POST(RouteBase, h.create)
 	admin.PUT(RouteBase+"/:id", h.update)
+
+	admin.GET(RouteMethod, h.method)
+}
+
+type methodResponse struct {
+	Method   string `json:"method"`
+	Provider string `json:"provider,omitempty"`
+}
+
+// swagger:route GET /method admin getMethod
+//
+// Get login method for a previously registered email address or username
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: loginMethod
+//       500: genericError
+func (h *Handler) method(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	// if the credentials can be looked up directly by email address then they're type password
+	_, _, err := h.r.PrivilegedIdentityPool().FindByCredentialsIdentifier(r.Context(), CredentialsTypePassword, ps.ByName("id"))
+	if err != nil {
+		// otherwise if they're OIDC credentials we have to find them by the verifiable address and then see if they are indeed OIDC
+		address, err := h.r.PrivilegedIdentityPool().FindVerifiableAddressByValue(r.Context(), VerifiableAddressTypeEmail, ps.ByName("id"))
+
+		if err != nil {
+			h.r.Writer().WriteError(w, r, err)
+			return
+		}
+
+		id, err := h.r.PrivilegedIdentityPool().GetIdentityConfidential(r.Context(), address.IdentityID)
+		if err != nil {
+			h.r.Writer().WriteError(w, r, err)
+			return
+		}
+
+		creds, ok := id.GetCredentials(CredentialsTypeOIDC)
+
+		var cfg map[string]interface{}
+		if err := json.NewDecoder(bytes.NewBuffer(creds.Config)).Decode(&cfg); err != nil {
```

Comment:
This will panic because ok is checked afterwards. It might also be simpler to use `gjson.Get(...)` instead of marshalling / type asserting

Reply:
Cool, thanks for the `gjson.Get()` suggestion, I will do it that way, definitely simpler

Reply:
I love that tool :)